src/fielddef.ts

```diff
@@ -156,6 +156,10 @@ export function isValueDef(channelDef: ChannelDef<any>): channelDef is ValueDef<
   return channelDef && 'value' in channelDef && channelDef['value'] !== undefined;
 }
 
+export function isScaleFieldDef(channelDef: ChannelDef<any>): channelDef is ScaleFieldDef<any> | PositionFieldDef<any> | LegendFieldDef<any, any> {
```

Comment:
Both `PositionFieldDef` and `LegendFieldDef` extends `ScaleFieldDef` -- so not sure if you really need to add `|` here.